### PR TITLE
Fix promise waiting for sqlite with `key` option

### PR DESCRIPTION
**Pull request by hansl**
If a `key` option is passed to the Sqlite driver, the connection promise does not wait on it and resolves on the first pragma. This PR makes sure that the Promise returned will wait for the key PRAGMA to be run.

Also took some time to simplify the reading of the createDatabaseConnection() function.

### CODE REVIEW

File: src/driver/sqlite/SqliteDriver.ts [hunk 1]
```diff
@@ -85,28 +85,36 @@ export class SqliteDriver extends AbstractSqliteDriver {
...
-                }
             });
-        });
+        }
+
+        // we need to enable foreign keys in sqlite to make sure all foreign key related features
+        // working properly. this also makes onDelete to work with sqlite.
+        await run(`PRAGMA foreign_keys = ON;`);
+
+        // in the options, if encryption key for for SQLCipher is setted.
```
Comment (pleerock) [inline src/driver/sqlite/SqliteDriver.ts]: `for for`

File: src/driver/sqlite/SqliteDriver.ts [hunk 2]
```diff
@@ -85,28 +85,36 @@ export class SqliteDriver extends AbstractSqliteDriver {
...
+        if (this.options.key) {
+            await run(`PRAGMA key = ${JSON.stringify(this.options.key)};`);
```
Comment (pleerock) [inline src/driver/sqlite/SqliteDriver.ts]: `this.options.key` looks weird. is 'pragma key' SQLCipher specific stuff? If yes, maybe option should be called `sqlCipherKey` ?

File: src/driver/sqlite/SqliteDriver.ts (hunk 1 above)
Comment (hansl) [reply src/driver/sqlite/SqliteDriver.ts]: Fixed the comment.

File: src/driver/sqlite/SqliteDriver.ts (hunk 2 above)
Comment (hansl) [reply src/driver/sqlite/SqliteDriver.ts]: This is code that's already released. I don't think changing the API is in spec for this PR.

Comment (pleerock) [reply src/driver/sqlite/SqliteDriver.ts]: oh I didn't realized that we already have this feature. Okay, thanks for the fix.

**Pull request merged**